numba/ir.py

```diff
@@ -186,6 +198,54 @@ def with_lineno(self, line, col=None):
 unknown_loc = Loc("unknown location", 0, 0)
 
 
+@total_ordering
+class SlotEqualityCheckMixin(object):
+    # some ir nodes are __dict__ free using __slots__ instead, this mixin
+    # should not trigger the unintended creation of __dict__.
+    __slots__ = tuple()
+
+    def __eq__(self, other):
+        if type(self) == type(other):
```

Comment:
I think the intention is to check for identical type.  Better use `is` instead of `==`.  The latter can be overloaded by metaclasses to do weird things.

Reply:
good point, will change.